influxql/parser.go

```diff
@@ -1508,6 +1508,8 @@ func (p *Parser) parseUnaryExpr() (Expr, error) {
 	case DURATION_VAL:
 		v, _ := ParseDuration(lit)
 		return &DurationLiteral{Val: v}, nil
+	case MUL:
+		return &Wildcard{}, nil
```

Comment:
I agree that we should do a lookahead for wildcards before parsing expressions but we can push that to a separate PR.